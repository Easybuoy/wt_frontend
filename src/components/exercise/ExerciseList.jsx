import React from "react";
import { Input } from "@chakra-ui/core";
import { ExercisesStyle } from "./ExerciseStyle";

function Exercises(props) {
  return (
    <div exercise-list-container>
      <ExercisesStyle>
        <nav></nav>
        <div className="search-container">
          <h4>Search</h4>
          <p>Find an exercise</p>
          <Input variant="filled" />
        </div>
        <div className="recommendations-container">
          <p>See recommended exercises</p>
          <p>These exercises were suggested based on your current progress</p>
        </div>
      </ExercisesStyle>
    </div>
  );
}

export default Exercises;
