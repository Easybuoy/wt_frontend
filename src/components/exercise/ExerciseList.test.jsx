import React from "react";
import { shallow } from "enzyme";

import ExerciseList from "./ExerciseList";

describe("<ExerciseList />", () => {
  let wrapper;

  const props = {
    location: { search: jest.mock() },
    client: { mutate: jest.fn(), query: jest.fn() },
    history: { push: jest.fn() }
  };

  beforeEach(() => {
    wrapper = shallow(<ExerciseList {...props} />);
  });

  it("renders the ExerciseList component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
