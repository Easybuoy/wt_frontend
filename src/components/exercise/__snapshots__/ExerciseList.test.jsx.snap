// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExerciseList /> renders the ExerciseList component correctly 1`] = `
<styled.div>
  <styled.div>
    <Search
      placeholder="Look for a specific exercise"
    />
    <withApollo(Exercises)
      exerciseName="Recommended Exercises"
      exerciseQuery="RECOMENDED_EXERCISES"
    />
    <withApollo(Exercises)
      exerciseName="Top Rated Exercises"
      exerciseQuery="TOP_RATED_EXERCISES"
    />
  </styled.div>
</styled.div>
`;
